### ISSUE TITLE
Question on improving detection

### ISSUE THREAD

**Issue by EvGe22** [AlexeyAB/darknet#1748]
[Here](https://github.com/AlexeyAB/darknet/blob/master/README.md#how-to-improve-object-detection) at the bottom it's stated that you can just change the size of the input to the network without retraining. Could you please explain how this is achieved?

**Comment by AlexeyAB** [comment]
The higher network resolution - the more information is kept in the image, since any image will be resized to the network resolution. I.e. the more resolution of image - the more details we can see on it.
Variety of relative sizes of objects in training-dataset and Data augmentation (jitter and random in cfg-file) allows us to use slightly higher resolution, than that was used for training.

Also about it: https://github.com/AlexeyAB/darknet/issues/1738#issuecomment-428315306

**Comment by EvGe22** [comment]
Thanks for the answer, I do get the stuff of **why** we should get a better performance, but I still don't quite get the underlying mechanism of feeding a bigger image into a network that was trained to process smaller ones. My knowledge of how NNs generally work tells me that it should not be practically possible without changing weights. 
The way I understand it: if we have weights for a function from an input layer 608x608x3 to some hidden layer NxNxM we cannot just change the size of the input layer to some other resolution and keep those weights since the tensor (or matrix) multiplication will just not work. How is this handled?

**Comment by AlexeyAB** [comment]
> since the tensor (or matrix) multiplication will just not work.

Why?
Only Fully-connected layers can't be resized, because number of weights depends on layer size. But any other layers can be resized: convolutional, maxpool, shortcut/residual, conctat/route, upsample/downsample, reorg, yolo/region, ...

----

Why do you think that if we trained model with network resolution 416x416, then it willn't work with resolution 448x448?

**Comment by EvGe22** [comment]
Yeah, changing the size of the input to the convolutional layer will still work, only the output will be of different size as well. I was previously thinking that Yolo had some fully connevted layers inside of it (not sure why). But after reading a bit more about its architecture, it turned out it does not(am I right?). That what was mainly bugging me. If we change the input size just a bit, then the whole pipeline will the end carrying that size difference. And if we had a fully connected layer that expects the input of specific size somewhere in down the line, it would break. 
So tell me please, if I understood this correctly:
In Yolo all layers do not depend on a specific size, they are either doing some 'filtering', detecting objects (this will still require some more digging, do not quite get how this works mathematically) and doing other interresting stuff, but no fully connected layers. So if we change the input from 416 to 448, our detections will just be ran on a bigger layer. I.e. 14x14 instead of 13x13.

**Comment by AlexeyAB** [comment]
Yes, this is correct. 
Yolo v1 had fully-connected layer, but Yolo v2 and v3 doesn't have it - so it can be used for any network size.

For example, if all our Training and Detection images have the same size 1000x1000, and if we **trained with network size 416x416** on Training-set with object sizes 20x20 - 50x50 pixels, and data augmentation increases this range to 10x10 - 100x100 pixels, then when we **detect objects with network size 832x832** - we can detect objects with sizes 5x5 - 50x50 by using this weights-file.
